transport/encode_decode.go

```diff
@@ -279,3 +307,18 @@ func isNameValid(name string) bool {
 	}
 	return nameRegex.MatchString(name)
 }
+
+func extractTarget(c echo.Context) (*domain.Target, error) {
+	var target *domain.Target
+	encodedTarget := c.Request().Header.Get(targetHeader)
+
+	if encodedTarget == "" {
+		return target, nil
```

Comment:
Kinda links in to above wondering if an empty target should return an error and then fallback to redis